lib/Persistence/AbstractPersistence.php

```diff
@@ -82,21 +82,23 @@ protected static function _exists($filename)
      */
     protected static function _initialize()
     {
-        // Create storage directory if it does not exist.
-        if (!is_dir(self::$_path)) {
-            if (!@mkdir(self::$_path, 0700)) {
-                throw new Exception('unable to create directory ' . self::$_path, 10);
+        if (property_exists($data->meta, 'webserver') && $data->meta->webserver && $this->_conf->getKey('webserver') == "Apache") {
```

Comment:
This now does not check the web server any more at all as far as I see… It just assumes "Apache by default" and only if the admin changes the settings in the options it assumes "any other web server".

However, I'd rather like to see this behaviour:
* By default the config is either empty or set to "auto" or so, which auto-selects/detects the correct server based on env. variables e.g.
* The server admin can override this behaviour by changing the option in the settings, so can e.g. force Apache or disallow it. This could be useful when the automatic detection does not work…